### PR TITLE
Packaging: make mac package require 10.12 or newer

### LINKED ISSUE #4290: Update mac package to error on unsupported version
Steps to reproduce
------------------
Install mac package on macOS 10.11

Expected behavior
-----------------

A friendly message that the package is only supported on 10.12 ( or similar)

Actual behavior
---------------

The package installs fine and errors when you run powershell

Details
----------------

https://developer.apple.com/library/content/documentation/DeveloperTools/Reference/DistributionDefinitionRef/Chapters/Distribution_XML_Ref.html#//apple_ref/doc/uid/TP40005370-CH100-SW33
https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man1/productbuild.1.html

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

Fixes #4290
Make mac package require 10.12 or newer.
Required that the package is installed to a disk with macOS installed.
The Apple example XML had a background image added as well, so I added a background image:
image

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [NA] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [NA] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [NA] [Add `[feature]` if the change is significant or affectes feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -745,7 +732,71 @@ function New-UnixPackage {
...
+    $newPackagePath = Join-Path $FpmPackage.DirectoryName $newPackageName
+
+    # -Force is not deleting the NewName if it exists, so delete it if it does
+    if ($Force -and (Test-Path -Path $newPackagePath))
+    {
+        Remove-Item -Force $newPackagePath
+    }
+
+    # Create the distribution xml
+    $distributionXmlPath = Join-Path -Path $tempDir -ChildPath '/powershellDistribution.xml'
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: I think the `/` is not required in `'/powershellDistribution.xml'`

File: tools/packaging/packaging.psm1 [hunk 2]
```diff
@@ -745,7 +732,71 @@ function New-UnixPackage {
...
+
+    # format distribution template with:
+    # 0 - title
+    # 1 - version
+    # 2 - package path
+    # 2 - minimum os version
+    $PackagingStrings.OsxDistributionTemplate -f "PowerShell - $Version", $Version, $packageName, '10.12' | Out-File -Encoding ascii -FilePath $distributionXmlPath -Force
+
+    #$FpmPackage = Rename-Item -Path $FpmPackage.FullName -NewName $newPackagePath -PassThru -ErrorAction Stop
```
Comment (adityapatwardhan) [inline tools/packaging/packaging.psm1]: Remove this line

**Comment by TravisEz13** [comment]
@thezim If you are familiar with this area, would you review this.  Look at the issue for the main reason for the change.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by adityapatwardhan — APPROVED** [review]

**Comment by thezim** [comment]
@TravisEz13 The package does not contain the launcher when I build your branch. Not sure if this is product of #5625 or I'm missing something in the build. Can you please verify if the launcher is there?

**Comment by TravisEz13** [comment]
@thezim Thanks for the review.  The issue was caused by the previous PR.  I have updated this PR.  Thanks.

File: tools/packaging/packaging.psm1 [hunk 3]
```diff
@@ -745,7 +733,70 @@ function New-UnixPackage {
...
+    # 1 - version
+    # 2 - package path
+    # 2 - minimum os version
+    $PackagingStrings.OsxDistributionTemplate -f "PowerShell - $Version", $Version, $packageName, '10.12' | Out-File -Encoding ascii -FilePath $distributionXmlPath -Force
+
+    log "Applying distribution.xml to package..."
+    Push-Location $tempDir
+    try
+    {
+        Start-NativeExecution -sb {productbuild --distribution $distributionXmlPath --resources $resourcesDir $newPackagePath}
```
Comment (daxian-dbw) [inline tools/packaging/packaging.psm1]: Non-blocking: it's better to add a comment explaining that `productbuild is a xcode command line tool which will be installed when homebrew is installed`.

File: tools/packaging/packaging.psm1 [hunk 4]
```diff
@@ -858,6 +920,12 @@ function Get-FpmArguments
         "$LinkSource=$LinkDestination"
     )
 
+    if($AppsFolder)
+    {
+        $Arguments += "$AppsFolder=/"
+    }
+
+
```
Comment (daxian-dbw) [inline tools/packaging/packaging.psm1]: Non-blocking: remove an extra new line.

**Review by daxian-dbw — COMMENTED** [review]
LGTM. Left two non-blocking comments which can be addressed in separate PRs.

**Pull request merged**